tensorflow/java/src/test/java/org/tensorflow/TensorMatcher.java

```diff
@@ -0,0 +1,86 @@
+/* Copyright 2016 The TensorFlow Authors. All Rights Reserved.
```

Comment:
I appreciate the value of matchers in general and how their use is a judgement call in any given situations. In this case, I would like to avoid additional indirection and maintenance burden.

```java
assertEquals(0, tensor.numDimensions());
assertEquals(0, tensor.shape().length());
```
very explicitly tells the reader what properties of the `Tensor` object are being validated.
On the other hand `assertThat(tensor, is(scalar()))` doesn't. Is it looking at all relevant properties? Which ones are relevant? I think on balance, I'd prefer the explicit calls here.